pkg/api/operators/operators_test.go

```diff
@@ -0,0 +1,264 @@
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	type args struct {
+		listers controller.Listers
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testInit",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	for _, tt := range tests {
+		// add a WLSOperator to one of the clusters
+		t.Run(tt.name, func(t *testing.T) {
+			// WHEN a WLS operator is appended to the cluster and the Init() method is invoked
+			createAndInsertWLSOperator(1, tt)
+			// ThEN the single operator with the expected name will be found in the Operators slice
+			assert.Equal(t, tt.listers, listerSet, "Wrong number of listers")
+			assert.Equal(t, 1, len(Operators), "Wrong number of WLS operators")
+			assert.Equal(t, "wls-operator-mytestbinding-1", Operators[0].Name, "Operator has the wrong name")
+		})
+	}
+}
+
+//TestReturnNoOperators ensures that an empty response is returned if no operators are available
+func TestReturnNoOperators(t *testing.T) {
+	// GIVEN an empty model binding pair is provided
+	Operators = nil
+	emptyModelBindingPairs := make(map[string]*types.ModelBindingPair)
+	Init(controller.Listers{
+		ModelBindingPairs: &emptyModelBindingPairs,
+	})
+	// setup http request processing
+	recorder := httptest.NewRecorder()
+	router := mux.NewRouter()
+	router.HandleFunc("/operators", ReturnAllOperators)
+	request, _ := http.NewRequest("GET", "/operators", nil)
+	// WHEN an HTTP request for the list of operators is submitted
+	router.ServeHTTP(recorder, request)
+	// THEN a HTTP OK response with an empty array body will be returned
+	assert.Equal(t, http.StatusOK, recorder.Code, "wrong response code")
+	var ops []Operator
+	err := json.Unmarshal(recorder.Body.Bytes(), &ops)
+	if err != nil {
+		t.Errorf("Error while reading response JSON: %s", err)
+	}
+	// assert empty response
+	assert.Empty(t, ops, "response should have no operators")
+}
+
+// TestReturnSingleOperator will test that a configured operator is returned
+func TestReturnSingleOperator(t *testing.T) {
+	// GIVEN an array of clusters and a model binding pair is provided
+	Operators = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	type args struct {
+		listers controller.Listers
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testReturnSingleOperator",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			createAndInsertWLSOperator(1, tt)
+			// setup http request processing
+			recorder := httptest.NewRecorder()
```

Comment:
what is this testing? not our code?

Reply:
???

It's invoking the API via the registered http handler, so yes - it's testing our code.